cpp/include/cudf/lists/filling.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <memory>
+
+namespace cudf::lists {
+/**
+ * @addtogroup lists_filling
+ * @{
+ * @file
+ * @brief Column APIs for individual list sequence
+ */
+
+/**
+ * @brief Create a lists column in which each row contains a sequence of values specified by a tuple
+ * of (`start`, `size`) parameters.
+ *
+ * Create a lists column in which each row is a sequence of values starting from a `start` value,
+ * incrementing by one, and its cardinality is specified by a `size` value. The `start` and `size`
+ * values used to generate each list is taken from the corresponding row of the input @p starts and
+ * @p sizes columns.
+ *
+ *  - @p sizes must be a column of integer types.
+ *  - If any row in the input columns is null then the corresponding output row will be null.
```

Comment:
Hm, my intuition would have been that these functions would require `starts.null_count() == 0` and `sizes.null_count() == 0`. I suspect that would simplify the implementation too.

Reply:
Thanks. `starts` and `sizes` would be the results of previous computation or data reading, thus they may be nullable. If we require `null_count == 0`,  the caller may spend extra time to filter out nulls.

On the other hand, implementation for nullable inputs just require an extra line to call `bitmask_and`. Thus, I argue to keep   input nullable.

Reply:
Okay, this seems reasonable. I'll update this.

Reply:
> as it is closer to the real world situations.

I don't think we have the data to make this claim. 

Making the inputs be non-nullable is more consistent with the rest of libcudf.

Reply:
In situations like these, the question you should ask yourself is what behavior is most fundamental? What if I know my inputs will always be non-nullable and don't want to pay any extra complexity or overhead to allow the inputs to be nullable. 

The existing behavior can be emulated with a more primitive function plus other functions if that's really the behavior you want.

Reply:
Sound good. I have updated the implementation. Thanks.